src/transformers/modeling_tf_utils.py

```diff
@@ -621,12 +631,11 @@ def generate(
         num_return_sequences = (
             num_return_sequences if num_return_sequences is not None else self.config.num_return_sequences
         )
-        decoder_start_token_id = (
-            decoder_start_token_id if decoder_start_token_id is not None else self.config.decoder_start_token_id
-        )
 
         if input_ids is not None:
-            batch_size = shape_list(input_ids)[0]  # overriden by the input batch_size
+            batch_size = input_ids.shape[0]  # overriden by the input batch_size
```

Comment:
I think we should not use `tensor.shape` in TF. The function `shape_list` was made so that `shape` behaves correctly in eager mode and no eager mode: https://github.com/huggingface/transformers/pull/3063